person.rb

```diff
@@ -21,6 +22,11 @@ def correct_name
     @name
   end
 
+  def add_rental(rental)
+    rental.person = self
+    @rentals.push(rental)
+  end
+

```

Comment:
- [ ] The `Rental` class is the go-between class between the `Person` and the `Book` classes. When creating the `Rental` class, in the `initialize` method, you set the `Person` and `Book` instances to the arguments passed and also update the instances' arrays. That is the correct implementation 👍. This means you don't need to repeat the same logic in the `add_rental` method of the `Person` class since the `Rental` class has that handled. 

In the `add_rental` method, all you have to do is add the `date` parameters and the `book` parameters, then create a new instance of the `Rental` class because the logic in the `initialize` method of the `Rental` class will handle the implementation of the many-to-many relationship between the `Book` and `Person` classes. 👍 

#### Hint:

```ruby
def add_rental(date, book)
  Rental.new(date, book, self)
end
```